src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Telemetry/VsTelemetryService.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Telemetry;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Telemetry
+{
+    [Export(typeof(ITelemetryService))]
+    [SuppressMessage("Microsoft.Cryptographic.Standard", "CA5350:MD5CannotBeUsed", Justification = "MD5 hash is not being used for security")]
+    internal class VsTelemetryService : ITelemetryService
```

Comment:
Yes, that's my last note above, his changes are not checked in so I pulled in part of his changes to attempt to minimize the merge conflict when things go in.